tests/api_tests/comments/test_views.py

```diff
@@ -0,0 +1,889 @@
+    def setUp(self):
+        super(TestCommentDetailView, self).setUp()
+        self.user = AuthUserFactory()
+        self.contributor = AuthUserFactory()
+        self.non_contributor = AuthUserFactory()
+
+        self.private_project = ProjectFactory(is_public=False, creator=self.user)
+        self.private_project.add_contributor(self.contributor, save=True)
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        self.public_project.add_contributor(self.contributor, save=True)
+        self.registration = RegistrationFactory(creator=self.user)
+
+        self.comment = CommentFactory(node=self.private_project, target=self.private_project, user=self.user)
+        self.public_comment = CommentFactory(node=self.public_project, target=self.public_project, user=self.user)
+        self.registration_comment = CommentFactory(node=self.registration, user=self.user)
+
+        self.private_url = '/{}comments/{}/'.format(API_BASE, self.comment._id)
+        self.public_url = '/{}comments/{}/'.format(API_BASE, self.public_comment._id)
+        self.registration_url = '/{}comments/{}/'.format(API_BASE, self.registration_comment._id)
+
+        self.payload = {
+            'data': {
+                'id': self.comment._id,
+                'type': 'comments',
+                'attributes': {
+                    'content': 'Updating this comment',
+                    'deleted': False
+                }
+            }
+        }
+
+        self.public_comment_payload = {
+            'data': {
+                'id': self.public_comment._id,
+                'type': 'comments',
+                'attributes': {
+                    'content': 'Updating this comment',
+                    'deleted': False
+                }
+            }
+        }
+
+    def test_private_node_logged_in_contributor_can_view_comment(self):
+        res = self.app.get(self.private_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.comment._id, res.json['data']['id'])
+
+    def test_private_node_logged_in_non_contributor_cannot_view_comment(self):
+        res = self.app.get(self.private_url, auth=self.non_contributor.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_private_node_logged_out_user_cannot_view_comment(self):
+        res = self.app.get(self.private_url, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    def test_public_node_logged_in_contributor_can_view_comment(self):
+        res = self.app.get(self.public_url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.public_comment._id, res.json['data']['id'])
+
+    def test_public_node_logged_in_non_contributor_can_view_comment(self):
+        res = self.app.get(self.public_url, auth=self.non_contributor.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.public_comment._id, res.json['data']['id'])
```

Comment:
I like that you confirm that it fetches the right comment- that makes the tests more specific and reliable. Thanks!